src/sage/schemes/projective/projective_morphism.py

```diff
@@ -600,19 +606,19 @@ def _matrix_times_polymap_(self, mat, h):
             sage: P.<x,y> = ProjectiveSpace(ZZ, 1)
             sage: H = Hom(P,P)
             sage: f = H([x^2 + y^2, y^2])
-            sage: matrix([[1,2], [0,1]]) * f
+            sage: matrix([[1,2], [0,1]]) * f                                            # needs sage.modules
             Scheme endomorphism of Projective Space of dimension 1 over Integer Ring
               Defn: Defined on coordinates by sending (x : y) to
                     (x^2 + 3*y^2 : y^2)
 
         ::
 
             sage: R.<x> = PolynomialRing(QQ)
-            sage: K.<i> = NumberField(x^2 + 1)                                                      # optional - sage.rings.number_field
-            sage: P.<x,y> = ProjectiveSpace(QQ, 1)                                                  # optional - sage.rings.number_field
-            sage: H = Hom(P, P)                                                                     # optional - sage.rings.number_field
-            sage: f = H([1/3*x^2 + 1/2*y^2, y^2])                                                   # optional - sage.rings.number_field
-            sage: matrix([[i,0], [0,i]]) * f                                                        # optional - sage.rings.number_field
+            sage: K.<i> = NumberField(x^2 + 1)                                          # needs sage.rings.number_field
+            sage: P.<x,y> = ProjectiveSpace(QQ, 1)
+            sage: H = Hom(P, P)
+            sage: f = H([1/3*x^2 + 1/2*y^2, y^2])
+            sage: matrix([[i,0], [0,i]]) * f                                            # needs sage.modules sage.rings.number_field
             Scheme endomorphism of Projective Space of dimension 1 over
```

Comment:
`i` defined in L617 is being used in below lines. Perhaps block-scoped tag? I wonder why `sage -fixdoctests` failed to detect this.